File: backend/src/main/java/net/alex9849/cocktailmaker/payload/dto/eventaction/DoNothingEventActionDto.java
package net.alex9849.cocktailmaker.payload.dto.eventaction;

import lombok.*;
import net.alex9849.cocktailmaker.model.eventaction.DoNothingEventAction;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class DoNothingEventActionDto {

    @NoArgsConstructor(access = AccessLevel.PRIVATE)
    public static class Request {
        @Getter @Setter @EqualsAndHashCode(callSuper = true)
        public static class Create extends EventActionDto.Request.Create {
        }
    }

    @NoArgsConstructor(access = AccessLevel.PRIVATE)
    public static class Response {
        @Getter @Setter @EqualsAndHashCode(callSuper = true)
        public static class Detailed extends EventActionDto.Response.Detailed {

            protected Detailed(DoNothingEventAction eventAction) {
                super(eventAction);
            }

            public String getType() {
                return "doNothing";
            }
        }
    }
}

File: frontend/src/models/dto/eventactions/DoNothingEventActionDto.ts
declare module 'DoNothingEventActionDto/Request' {
  import EventActionDtoRequest from 'EventActionDto/Request'

  export interface Create extends EventActionDtoRequest.Create {
    type: 'doNothing'
  }
}

declare module 'DoNothingEventActionDto/Response' {
  import EventActionDtoResponse from 'EventActionDto/Response'

  export interface Detailed extends EventActionDtoResponse.Detailed {
    type: 'doNothing'
  }
}
